xarray/tests/test_combine.py

```diff
@@ -959,7 +1017,7 @@ def test_combine_by_coords_combine_attrs_variables(
     def test_infer_order_from_coords(self):
         data = create_test_data()
         objs = [data.isel(dim2=slice(4, 9)), data.isel(dim2=slice(4))]
-        actual = combine_by_coords(objs)
+        actual = combine_by_coords(objs, data_vars="all")
```

Comment:
This one should not require data_vars="all", so we have a false positive warning. That should only be needed if we are concatenating along a new dimension, but here we are concatenating along an existing dimension: dim2